File: SeleniumTests/src/main/java/Selenium/Common/Helpers/CustomImplicitTimeout.java
package Selenium.Common.Helpers;

import Selenium.Common.Configs.Configuration;
import org.openqa.selenium.WebDriver;

import java.util.concurrent.TimeUnit;

public class CustomImplicitTimeout implements AutoCloseable {
    private WebDriver webDriver;
    private final Configuration config = Configuration.getInstance();
    private final Integer defaultTimeout = Integer.parseInt(config.getValue("implicitTimeout"));

    public CustomImplicitTimeout(WebDriver webDriver, Integer timeout) {
        this.webDriver = webDriver;
        webDriver.manage().timeouts().implicitlyWait(timeout, TimeUnit.SECONDS);
    }

    @Override
    public void close() {
        webDriver.manage().timeouts().implicitlyWait(defaultTimeout, TimeUnit.SECONDS);
    }
}

File: C#/SeleniumTest/SeleniumTest/Core/CustomImplicitTimeout.cs
using OpenQA.Selenium;
using System;

namespace SeleniumTest.Core
{
    public class CustomImplicitTimeout : IDisposable
    {
        private IWebDriver WebDriver { get; set; }
        private static readonly Configuration config = Configuration.GetInstance();
        private static int defaultTimeout = int.Parse(config.GetValue("implicitTimeout"));

        public CustomImplicitTimeout(IWebDriver webDriver, int timeout)
        {
            WebDriver = webDriver;
            WebDriver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(timeout);
        }

        public void Dispose()
        {
            WebDriver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(defaultTimeout);
        }
    }
}
